### PR TITLE
Map loader for official maps (Supports ANY Map version)

### PR DESCRIPTION
**Pull request by OrN**
Load official maps into the server with "based_map_data" config option. The default I think is 64, which should be the latest.

There are 2 different compression methods used, first is BZip and is used up to maps27. BZip2 is then used all the way to 64.
There are also 3 different map formats. The first one is .jm in the maps archive. The second one is .dat and .hei combo, but .dat is in a different format (Eggsampler 127 has it). The third is what we use in 177/234/235. This PR can handle all of them.

### CODE REVIEW

File: server/src/com/openrsc/server/io/Tile.java
Comment (Hubcapp) [inline server/src/com/openrsc/server/io/Tile.java:17]: empty comment

File: server/src/com/openrsc/server/io/WorldLoader.java
Comment (Hubcapp) [inline server/src/com/openrsc/server/io/WorldLoader.java:50]: This is fine, but maybe would have been better to be explicit & read some property of JContent for what version of the map the jagArchive contains. 👍

**Review by Hubcapp — COMMENTED** [review]
Will want to test & would say BZLib & BZState are close to completely obfuscated, but if it works, then it's good.

File: server/src/com/openrsc/server/io/WorldLoader.java
Comment (OrN) [reply server/src/com/openrsc/server/io/WorldLoader.java:50]: We can't afaik, the content files have used the same format since the beggining of time. Obfuscated BZLib is an implementation based on bzip back in the day they used, I know they use BZip2 for a fact in later versions, and we will need to check for that based on config most likely of what the user say version they are using.

Comment (Hubcapp) [reply server/src/com/openrsc/server/io/WorldLoader.java:50]: Yeah, that works, sounds good.

**Review by lmsv-mx123 — APPROVED** [review]
Satisfied with PR functionality wise, checked in particular the changed areas (ghost town, POH area, and some parts of Varrock and Falador that changed, to ensure there were no blocked tiles walking). Checked also general map14 walking in unchanged areas and dungeons. Also checked projectile shooting when its around bed, gates and fences and was proper as well. Finally checked that player could not walk into non map14 areas (like wild) on modern client.

**Review by Hubcapp — APPROVED** [review]

**Comment by Hubcapp** [comment]
good to merge unless we're waiting on maps28+ support

**Comment by Hubcapp** [comment]
We are in fact waiting for maps28+ support in this same PR. Do not merge.

**Review by Hubcapp — CHANGES_REQUESTED** [review]
maps28+

**Comment by lmsv-mx123** [comment]
Checked on map34 and map64 and looks quite solid. Great work

**Review by lmsv-mx123 — CHANGES_REQUESTED** [review]
With latest is not loading map14 tile data for me and also giving this error:
``` [java] java.lang.ArrayIndexOutOfBoundsException: Index 20736 out of bounds for length 20736
     [java]     at com.openrsc.server.io.JContent.unpack(JContent.java:71) ~[core.jar:?]
     [java]     at com.openrsc.server.io.WorldLoader.loadJAGSector(WorldLoader.java:65) ~[core.jar:?]
     [java]     at com.openrsc.server.io.WorldLoader.loadSection(WorldLoader.java:343) ~[core.jar:?]
     [java]     at com.openrsc.server.io.WorldLoader.loadWorld(WorldLoader.java:512) ~[core.jar:?]
     [java]     at com.openrsc.server.model.world.region.RegionManager.load(RegionManager.java:30) ~[core.jar:?]
     [java]     at com.openrsc.server.model.world.World.load(World.java:348) ~[core.jar:?]
     [java]     at com.openrsc.server.Server.start(Server.java:329) ~[core.jar:?]
     [java]     at com.openrsc.server.Server.startServer(Server.java:132) ~[core.jar:?]
     [java]     at com.openrsc.server.Server.main(Server.java:188) ~[core.jar:?]```

**Comment by lmsv-mx123** [comment]
> With latest is not loading map14 tile data for me and also giving this error:
> 
> ```
>      [java]     at com.openrsc.server.io.JContent.unpack(JContent.java:71) ~[core.jar:?]
>      [java]     at com.openrsc.server.io.WorldLoader.loadJAGSector(WorldLoader.java:65) ~[core.jar:?]
>      [java]     at com.openrsc.server.io.WorldLoader.loadSection(WorldLoader.java:343) ~[core.jar:?]
>      [java]     at com.openrsc.server.io.WorldLoader.loadWorld(WorldLoader.java:512) ~[core.jar:?]
>      [java]     at com.openrsc.server.model.world.region.RegionManager.load(RegionManager.java:30) ~[core.jar:?]
>      [java]     at com.openrsc.server.model.world.World.load(World.java:348) ~[core.jar:?]
>      [java]     at com.openrsc.server.Server.start(Server.java:329) ~[core.jar:?]
>      [java]     at com.openrsc.server.Server.startServer(Server.java:132) ~[core.jar:?]
>      [java]     at com.openrsc.server.Server.main(Server.java:188) ~[core.jar:?]```
> ```

Fixed now on latest

**Review by lmsv-mx123 — APPROVED** [review]

**Pull request merged**
